python/rikai/spark/functions/vision.py

```diff
@@ -122,9 +125,13 @@ def video_to_images(
         The sampling rate in number of frames
     max_samples : Int
         Yield at most this many frames (-1 means no max)
+     youtube_uri: Str
```

Comment:
nit: str is not capitalized

1. The way you have it for VideoStream it's still writing to local disk on each spark executor at the point of invocation. You'll want to make sure that in both YouTubeVideo and VideoStream cases you're writing to the supplied argument (maybe an S3 path in real usage).
2. Instead of a `youtube_uri` argument, you want to call this `base_path` or `output_dir` or something similar. Since that's what it actually is.
3. don't make it optional for now -- the tempdir is cleared as soon as the context exists

Reply:
I think it is a good time now to think a bit about the user experience of how users interact with the directory to store assets. 

Should we expect a global asset directory (i.e., `rikai.option.set("rikai.asset.dirctory", "s3://bucket/foo/assets"`) or ask users to pass `lit(”s3://bucket/foo/assets")` each time?